internal/pkg/agent/application/actions/handlers/handler_action_unenroll.go

```diff
@@ -60,11 +72,46 @@ func (h *Unenroll) Handle(ctx context.Context, a fleetapi.Action, acker acker.Ac
 		return fmt.Errorf("invalid type, expected ActionUnenroll and received %T", a)
 	}
 
+	// Find inputs that want to receive UNENROLL action
+	// Endpoint needs to receive a signed UNENROLL action in order to be able to uncontain itself
+	state := h.coord.State()
+	comps, units := findMatchingUnitsByActionType(state, a.Type())
+	if len(comps) > 0 {
+		// Iterate through founc components and forward the UNENROLL action
+		for i, comp := range comps {
+			unit := units[i]
+			// Deserialize the action into map[string]interface{} for dispatching over to the apps
+			params, err := action.MarshalMap()
+			if err != nil {
+				return err
+			}
+
+			h.log.Debugf("Dispatch %v action to %v", a.Type(), unit.Config.Type)
+
+			res, err := h.coord.PerformAction(ctx, comp, unit, unit.Config.Type, params)
+
+			// If Endpoint UNENROLL fails, continue with Agent UNENROLL
```

Comment:
Does it make more sense to mark the action as failed with an error here instead?

Reply:
Sure, can return the error here instead